dysnix/app/templates/pdb.yaml

```diff
@@ -0,0 +1,43 @@
+{{- define "app.resources.pdb" -}}
+{{- if .Values.pdb.create }}
+---
+apiVersion: {{ include "common.capabilities.policy.apiVersion" . }}
+kind: PodDisruptionBudget
+metadata:
+  name: {{ include "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    {{- include "app.labels.component" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- if .Values.commonAnnotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+spec:
+{{- with .Values.pdb }}
+{{- if and ( hasKey . "minAvailable" ) (not (eq .minAvailable "") ) }}
```

Comment:
Comparing different data types with `eq` method is not supported.

```
Error: template: app/templates/pdb.yaml:41:8: executing "app/templates/pdb.yaml" at <include "app.pdb" (dict "component" $component "values" $values "top" $)>: error calling include: template: app/templates/_resources.tpl:92:6: executing "app.pdb" at <include "app.resources.include" (dict "resource" "pdb" | merge .)>: error calling include: template: app/templates/_resources.tpl:48:6: executing "app.resources.include" at <include (printf "app.resources.%s" .resource) $context>: error calling include: template: app/templates/pdb.yaml:20:45: executing "app.resources.pdb" at <eq .minAvailable "">: error calling eq: incompatible types for comparison
````

WDYT on `kindIs` instead of `eq` check? F.e.:

`(not (kindIs "string" .maxUnavailable))`

Reply:
Implemented, had to use `float64` instead of `int` due to [helm issue](https://github.com/helm/helm/issues/1707)